src/momento/config/configuration.py

```diff
@@ -106,10 +131,60 @@ def with_root_certificates_pem(self, root_certificates_pem_path: Path) -> Config
             root_certificates_pem_path (Path): the new root certificates.
 
         Returns:
-            ConfigurationBase: the new Configuration.
+            Configuration: the new Configuration.
         """
         grpc_configuration = self._transport_strategy.get_grpc_configuration().with_root_certificates_pem(
             root_certificates_pem_path
         )
         transport_strategy = self._transport_strategy.with_grpc_configuration(grpc_configuration)
         return self.with_transport_strategy(transport_strategy)
+
+    def with_middlewares(self, middlewares: List[Middleware]) -> Configuration:
```

Comment:
I believe the intent of this is to *replace* the existing middleware list instead of extending it?